### PR TITLE
FEAT: `ldexp` ufunc implementation

### PR DESCRIPTION
**Pull request by SwayamInSync**
## Copilot Summary
This pull request adds full support for the `ldexp` function (computing `x * 2**exp`) for quad precision types in the codebase. It introduces the mixed-type operation for quad and integer types, integrates the operation as a NumPy ufunc, and provides comprehensive tests to ensure correctness and edge case handling. The release tracker is updated to reflect this new support.

Quad precision `ldexp` implementation:

* Added mixed-type operation functions `quad_ldexp` and `ld_ldexp` for quad and long double types, including proper handling of NaN, zero, and infinity cases in `ops.hpp`.

NumPy ufunc integration:

* Implemented descriptor resolution and strided loop logic for the quad precision `ldexp` ufunc, and registered the ufunc with proper type promotion in `binary_ops.cpp`.
* Registered the new `ldexp` ufunc in `init_quad_binary_ops`, making it available for use.

Testing and documentation:

* Added a comprehensive test suite for the quad precision `ldexp` function, covering basic functionality, zero, infinity, NaN, overflow, and underflow cases in `test_quaddtype.py`.
* Updated the release tracker to mark `ldexp` as implemented and tested for quad precision.

### CODE REVIEW

**Comment by SwayamInSync** [comment]
one more thing to note is that on these systems long double is 128 bit hence Sleef_quad is also getting mapped to long double

File: quaddtype/numpy_quaddtype/src/ops.hpp [hunk 1]
```diff
@@ -1047,6 +1047,62 @@ quad_spacing(const Sleef_quad *x)
     return result;
 }
 
+// Mixed-type operations (quad, int32) -> quad
```
Comment (juntyr) [inline quaddtype/numpy_quaddtype/src/ops.hpp]:
```suggestion
// Mixed-type operations (quad, int) -> quad
```

File: quaddtype/numpy_quaddtype/src/ops.hpp [hunk 2]
```diff
@@ -1047,6 +1047,64 @@ quad_spacing(const Sleef_quad *x)
...
+        return *x;
+    }
+    
+    // ±inf * 2^exp = ±inf (preserves sign of infinity)
+    if (quad_isinf(x)) {
+        return *x;
+    }
+    
+    // Pass dereferenced values directly to SLEEF (no conversion needed)
+    Sleef_quad result = Sleef_ldexpq1(*x, *exp);
```
[the remark this answers is not in the document]
Comment (juntyr) [reply quaddtype/numpy_quaddtype/src/ops.hpp]: Did you confirm that we still need those manual special cases?

**Comment by juntyr** [comment]
@SwayamInSync Ok, feel free to merge once CI is green
